controllers/setup.go

```diff
@@ -0,0 +1,89 @@
+package controllers
+
+import (
+	"context"
+	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"path/filepath"
+
+	"kubevirt.io/ssp-operator/internal/operands"
+	"kubevirt.io/ssp-operator/internal/operands/common-templates"
+	"kubevirt.io/ssp-operator/internal/operands/data-sources"
+	"kubevirt.io/ssp-operator/internal/operands/metrics"
+	"kubevirt.io/ssp-operator/internal/operands/node-labeller"
+	"kubevirt.io/ssp-operator/internal/operands/template-validator"
+	"sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+func CreateAndSetupReconciler(mgr controllerruntime.Manager) error {
+	templatesFile := filepath.Join(templateBundleDir, "common-templates-"+common_templates.Version+".yaml")
+	templatesBundle, err := common_templates.ReadTemplates(templatesFile)
+	if err != nil {
+		return err
+	}
+
+	sspOperands := []operands.Operand{
+		metrics.New(),
+		template_validator.New(),
+		common_templates.New(templatesBundle),
+		data_sources.New(),
+		node_labeller.New(),
+	}
+
+	var requiredCrds []string
+	for i := range sspOperands {
+		requiredCrds = append(requiredCrds, sspOperands[i].RequiredCrds()...)
+	}
+
+	// Check if all needed CRDs exist
+	crdList := &extv1.CustomResourceDefinitionList{}
+	err = mgr.GetAPIReader().List(context.TODO(), crdList)
```

Comment:
wouldn't be possible to use `ListOptions` with `FieldSelector` (use field selector for CRD names)? And list only CRDS which are interesting for us? And with that result it should be possible to easily check is all CRDs are ready `if len(crdList.Items) == len(requiredCrds)`

Reply:
No, it is not possible. 
The field selector does not support set operations like label selector, so we cannot use: 
```
metadata.name IN {"datasources.cdi.kubevirt.io", "dataimportcrons.cdi.kubevirt.io"}
```
nor does it support logical `OR`, like:
```
(metadata.name == "datasources.cdi.kubevirt.io") OR (metadata.name == "dataimportcrons.cdi.kubevirt.io")
```